core/src/main/java/org/elasticsearch/index/query/BoolQueryParser.java

```diff
@@ -78,12 +78,12 @@ public BoolQueryBuilder fromXContent(QueryParseContext parseContext) throws IOEx
                     shouldClauses.add(query);
                     break;
                 case "filter":
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
```

Comment:
yes, the filter flag is now handled in the QueryShardContext when actually creating the lucene queries. Checked for this case that we do so.